src/Http/Offer/PatchOfferRequestHandler.php

```diff
@@ -5,62 +5,60 @@
 namespace CultuurNet\UDB3\Http\Offer;
 
 use Broadway\CommandHandling\CommandBus;
-use CultuurNet\UDB3\Offer\OfferType;
-use CultuurNet\UDB3\HttpFoundation\Response\NoContent;
-use DateTime;
-use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\Response;
+use CultuurNet\UDB3\Http\Request\RouteParameters;
+use CultuurNet\UDB3\Http\Response\NoContentResponse;
+use CultuurNet\UDB3\Json;
 use CultuurNet\UDB3\StringLiteral;
+use DateTime;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
 
-class PatchOfferRestController
+final class PatchOfferRequestHandler implements RequestHandlerInterface
 {
     public const DOMAIN_MODEL_REGEX = '/.*domain-model=([a-zA-Z]*)/';
 
     private CommandBus $commandBus;
 
-    private OfferType $offerType;
-
-    public function __construct(
-        OfferType $offerType,
-        CommandBus $commandBus
-    ) {
-        $this->offerType = $offerType;
+    public function __construct(CommandBus $commandBus)
+    {
         $this->commandBus = $commandBus;
     }
 
-    public function handle(Request $request, string $cdbid): Response
+    public function handle(ServerRequestInterface $request): ResponseInterface
     {
+        $routeParameters = new RouteParameters($request);
+        $offerType = $routeParameters->getOfferType();
+        $offerId = $routeParameters->getOfferId();
+
         $domainModel = $this->parseDomainModelNameFromRequest($request);
-        $commandClass = 'CultuurNet\UDB3\\' . $this->offerType->toString() . '\Commands\Moderation\\' . $domainModel;
+        $commandClass = 'CultuurNet\UDB3\\' . $offerType->toString() . '\Commands\Moderation\\' . $domainModel;
```

Comment:
I know it's not new code but I'm curious, why are some backslashes escaped and others not in this string? Is the escaping even necessary in single quotes?

Reply:
Otherwise if we do `\'` then it will escape the single quote, so we need `\\'` to avoid escaping

Reply:
Ah yes I see now! Thanks for explaining :)